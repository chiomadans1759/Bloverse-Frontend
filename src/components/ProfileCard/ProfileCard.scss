@import 'sass-rem';

.profile-img {
  width: rem(64px);
  height: rem(64px);
}

.profile-name {
  font-size: rem(18px);
  font-weight: 700;
}

.avatar {
  display: flex;
  justify-content: end;
  align-items: baseline;
}

.profile-edit-icon {
  margin-top: rem(-88px);
  color: #ffffff;
  font-weight: 100;
}

.profile-body {
  color: #848384;
}

main.card {
  border-radius: 0 !important;
  .card-img-top {
    border-top-left-radius: 0;
    border-top-right-radius: 0;
  }
  span>svg.back-arrow {
    color: #ffffff;
  }
}

#number {
  border: rem(1px) solid #3d98eb !important;
  color: #3d98eb;
  cursor: pointer;
}

.profile {
  &__text {
    font-size: rem(16px);
    font-weight: 500;
    color: #6c6c6c;
    font-family: Montserrat;
  }

  p {
    font-size: rem(16px);
  }

  a {
    font-size: rem(14px);
  }

  .card-footer {
    font-size: rem(14px);
  }

  .interests {
    .category-text {
      font-size: rem(12px);
    }
  }
  .social {
    .social-icon {
      height: 14px;
      width: 14px;
    }
    .social-text {
      font-size: rem(12px);
    }
  }

  .followers-icon {
    background-color: #096DD9;
    width: rem(22px);
    height: rem(16px);
    color: #ffffff;
  }
}

.footer-color {
  color: rgba(0, 0, 0, .5);
}

@media screen and (min-width: 700px) {
  .profile-img {
    width: rem(64px);
    height: rem(64px);
  }
  
  .profile-name {
    font-size: rem(20px);
    font-weight: 700;
  }
  
  .avatar {
    display: flex;
    justify-content: end;
    align-items: baseline;
  }
  
  .profile-edit-icon {
    margin-top: rem(-88px);
    color: #ffffff;
    font-weight: 100;
  }
  
  .profile-body {
    color: #848384;
  }

  main.card {
    border-radius: rem(4px) !important;
    .card-image-top {
      border-top-left-radius: calc(.25rem - 1px);
      border-top-right-radius: calc(.25rem - 1px);
    }
  }

  #number {
    border: rem(1px) solid #3d98eb !important;
    color: #3d98eb;
    cursor: pointer;
  }
  
  .profile {
    &__text {
      font-size: rem(16px);
      font-weight: 500;
      color: #6c6c6c;
      font-family: Montserrat;
    }
  
    p {
      font-size: rem(16px);
    }
  
    a {
      font-size: rem(14px);
    }
  
    .card-footer {
      font-size: rem(14px);
    }
  }
}
