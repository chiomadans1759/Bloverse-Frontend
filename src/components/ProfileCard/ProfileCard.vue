<template>
  <main class="card profile">
    <img src="@/assets/Avatar.png" class="card-img-top" alt="...">
    <div class="profile-edit-icon d-flex justify-content-end p-3">
      <i class="fal fa-user-edit fa-2x"></i>
    </div>
    <div class="avatar mx-4">
      <img src="@/assets/Avatar2.png" class="profile-img rounded" alt="...">
      <span class="profile-name pl-3">Johndoe</span>
    </div>
    <div class="card-body profile-body">
      <h5 class="card-title">John Doe</h5>
      <p>
        <span class="pr-2 pb-5">
          <i class="fal fa-phone"></i>
        </span> +234 000 000 0000
      </p>
      <p>
        <span class="pr-2 py-1">
          <i class="fal fa-envelope"></i>
        </span> email@bloverse.com
      </p>
      <p>
        <span class="pr-2">
          <i class="fal fa-map-marker-alt"></i>
        </span> Lagos, Nigeria
      </p>
      <div class="py-4 profile-content">
        <p
          class="profile__text"
        >Some quick example text to build on the card title and make up the bulk of the card's content.</p>
      </div>
      <div class="follow d-flex justify-content-between">
        <a data-toggle="modal" href="#following-modal">Following</a>
        <span class="border rounded px-1 mr-2" data-toggle="modal" href="#following-modal" id="number">5</span>
      </div>
    </div>
    <div class="card-footer text-muted bg-white">Member Since:
      <strong>December, 2018</strong>
    </div>
    <modal target="following-modal" title="Following" hideFooter>
      <div class="bl-modal-content">
        <div class="row align-items-center mt-2 mb-2">
          <div class="col-3 col-md-2 d-flex justify-content-center">
            <img class="user-image img-fluid" src="@/assets/smiling-author.jpg">
          </div>
          <div class="col-9 col-sm-9 col-md-6 pl-0">
            <p class="username">
              JohnDoe
              <span class="sub-span point d-inline d-md-none">
                <a href="#">Unfollow</a>
              </span>
            </p>
            <p class="sub-text">
              <span class="sub-span">24</span> Followers
              <span class="sub-span point">234</span> Articles
            </p>
          </div>
          <div class="d-none d-sm-none col-md-4 d-md-flex justify-content-end">
            <button type="button" class="btn btn-outline-primary">Following</button>
          </div>
        </div>
        <div class="row align-items-center mt-2 mb-2">
          <div class="col-3 col-md-2 d-flex justify-content-center">
            <img class="user-image img-fluid" src="@/assets/smiling-author.jpg">
          </div>
          <div class="col-9 col-sm-9 col-md-6 pl-0">
            <p class="username">
              JohnDoe
              <span class="sub-span point d-inline d-md-none">
                <a href="#">Unfollow</a>
              </span>
            </p>
            <p class="sub-text">
              <span class="sub-span">24</span> Followers
              <span class="sub-span point">234</span> Articles
            </p>
          </div>
          <div class="d-none d-sm-none col-md-4 d-md-flex justify-content-end">
            <button type="button" class="btn btn-outline-primary">Following</button>
          </div>
        </div>
        <div class="row align-items-center mt-2 mb-2">
          <div class="col-3 col-md-2 d-flex justify-content-center">
            <img class="user-image img-fluid" src="@/assets/smiling-author.jpg">
          </div>
          <div class="col-9 col-sm-9 col-md-6 pl-0">
            <p class="username">
              JohnDoe
              <span class="sub-span point d-inline d-md-none">
                <a href="#">Unfollow</a>
              </span>
            </p>
            <p class="sub-text">
              <span class="sub-span">24</span> Followers
              <span class="sub-span point">234</span> Articles
            </p>
          </div>
          <div class="d-none d-sm-none col-md-4 d-md-flex justify-content-end">
            <button type="button" class="btn btn-outline-primary">Following</button>
          </div>
        </div>
      </div>
    </modal>
  </main>
</template>

<script>
import Modal from "@/components/Modal/Modal";

export default {
  name: "ProfileCard",
  components: {
    Modal
  }
};
</script>

<style lang="scss" scoped>
@import "./ProfileCard";
</style>
