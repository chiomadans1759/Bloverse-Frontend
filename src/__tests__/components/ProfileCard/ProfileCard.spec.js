import { mount, RouterLinkStub } from '@vue/test-utils';
import ProfileCard from '@/components/ProfileCard/ProfileCard.vue';

describe('ProfileCard Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(ProfileCard, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
