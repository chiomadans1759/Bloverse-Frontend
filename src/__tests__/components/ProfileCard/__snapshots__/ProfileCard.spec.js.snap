// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard Component should render correctly 1`] = `
<main
  class="card profile"
>
  <img
    alt="..."
    class="card-img-top"
    src="@/assets/Avatar.png"
  />
   
  <div
    class="profile-edit-icon d-flex justify-content-end p-3"
  >
    <i
      class="fal fa-user-edit fa-2x"
    />
  </div>
   
  <div
    class="avatar mx-4"
  >
    <img
      alt="..."
      class="profile-img rounded"
      src="@/assets/Avatar2.png"
    />
     
    <span
      class="profile-name pl-3"
    >
      Johndoe
    </span>
  </div>
   
  <div
    class="card-body profile-body"
  >
    <h5
      class="card-title"
    >
      John Doe
    </h5>
     
    <p>
      <span
        class="pr-2 pb-5"
      >
        <i
          class="fal fa-phone"
        />
      </span>
       +234 000 000 0000
    
    </p>
     
    <p>
      <span
        class="pr-2 py-1"
      >
        <i
          class="fal fa-envelope"
        />
      </span>
       email@bloverse.com
    
    </p>
     
    <p>
      <span
        class="pr-2"
      >
        <i
          class="fal fa-map-marker-alt"
        />
      </span>
       Lagos, Nigeria
    
    </p>
     
    <div
      class="py-4 profile-content"
    >
      <p
        class="profile__text"
      >
        Some quick example text to build on the card title and make up the bulk of the card's content.
      </p>
    </div>
     
    <div
      class="follow d-flex justify-content-between"
    >
      <a
        data-toggle="modal"
        href="#following-modal"
      >
        Following
      </a>
       
      <span
        class="border rounded px-1 mr-2"
        data-toggle="modal"
        href="#following-modal"
        id="number"
      >
        5
      </span>
    </div>
  </div>
   
  <div
    class="card-footer text-muted bg-white"
  >
    Member Since:
    
    <strong>
      December, 2018
    </strong>
  </div>
   
  <main
    id="bl-modal"
  >
    <div
      aria-hidden="true"
      aria-labelledby="customModal"
      class="modal fade"
      id="following-modal"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
              id="customModal"
            >
              Following
            </h5>
             
            <button
              aria-label="Close"
              class="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
           
          <div
            class="modal-body"
          >
            <div
              class="bl-modal-content"
            >
              <div
                class="row align-items-center mt-2 mb-2"
              >
                <div
                  class="col-3 col-md-2 d-flex justify-content-center"
                >
                  <img
                    class="user-image img-fluid"
                    src="@/assets/smiling-author.jpg"
                  />
                </div>
                 
                <div
                  class="col-9 col-sm-9 col-md-6 pl-0"
                >
                  <p
                    class="username"
                  >
                    
            JohnDoe
            
                    <span
                      class="sub-span point d-inline d-md-none"
                    >
                      <a
                        href="#"
                      >
                        Unfollow
                      </a>
                    </span>
                  </p>
                   
                  <p
                    class="sub-text"
                  >
                    <span
                      class="sub-span"
                    >
                      24
                    </span>
                     Followers
            
                    <span
                      class="sub-span point"
                    >
                      234
                    </span>
                     Articles
          
                  </p>
                </div>
                 
                <div
                  class="d-none d-sm-none col-md-4 d-md-flex justify-content-end"
                >
                  <button
                    class="btn btn-outline-primary"
                    type="button"
                  >
                    Following
                  </button>
                </div>
              </div>
               
              <div
                class="row align-items-center mt-2 mb-2"
              >
                <div
                  class="col-3 col-md-2 d-flex justify-content-center"
                >
                  <img
                    class="user-image img-fluid"
                    src="@/assets/smiling-author.jpg"
                  />
                </div>
                 
                <div
                  class="col-9 col-sm-9 col-md-6 pl-0"
                >
                  <p
                    class="username"
                  >
                    
            JohnDoe
            
                    <span
                      class="sub-span point d-inline d-md-none"
                    >
                      <a
                        href="#"
                      >
                        Unfollow
                      </a>
                    </span>
                  </p>
                   
                  <p
                    class="sub-text"
                  >
                    <span
                      class="sub-span"
                    >
                      24
                    </span>
                     Followers
            
                    <span
                      class="sub-span point"
                    >
                      234
                    </span>
                     Articles
          
                  </p>
                </div>
                 
                <div
                  class="d-none d-sm-none col-md-4 d-md-flex justify-content-end"
                >
                  <button
                    class="btn btn-outline-primary"
                    type="button"
                  >
                    Following
                  </button>
                </div>
              </div>
               
              <div
                class="row align-items-center mt-2 mb-2"
              >
                <div
                  class="col-3 col-md-2 d-flex justify-content-center"
                >
                  <img
                    class="user-image img-fluid"
                    src="@/assets/smiling-author.jpg"
                  />
                </div>
                 
                <div
                  class="col-9 col-sm-9 col-md-6 pl-0"
                >
                  <p
                    class="username"
                  >
                    
            JohnDoe
            
                    <span
                      class="sub-span point d-inline d-md-none"
                    >
                      <a
                        href="#"
                      >
                        Unfollow
                      </a>
                    </span>
                  </p>
                   
                  <p
                    class="sub-text"
                  >
                    <span
                      class="sub-span"
                    >
                      24
                    </span>
                     Followers
            
                    <span
                      class="sub-span point"
                    >
                      234
                    </span>
                     Articles
          
                  </p>
                </div>
                 
                <div
                  class="d-none d-sm-none col-md-4 d-md-flex justify-content-end"
                >
                  <button
                    class="btn btn-outline-primary"
                    type="button"
                  >
                    Following
                  </button>
                </div>
              </div>
            </div>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
  </main>
</main>
`;
